coilsnake/modules/eb/AnimationModule.py

```diff
@@ -0,0 +1,194 @@
+        self.unknown = unknown
+
+        if graphics_data_size:
+            num_tiles = EbGraphicTileset.tiles_from_parameters(graphics_data_size, TILE_WIDTH, TILE_HEIGHT, TILE_BPP)
+        else: # Make tileset with maximum number of tiles possible for each frame of animation
+            num_tiles = SCREEN_WIDTH_TILES * SCREEN_HEIGHT_TILES * frames
+
+        # Animations are 2 bpp, so the palette will have 4 colors
+        self.palette = EbPalette(num_subpalettes=1, subpalette_length=4)
+        self.graphics = EbGraphicTileset(num_tiles=num_tiles, tile_width=TILE_WIDTH, tile_height=TILE_HEIGHT)
+        self.arrangements = [EbTileArrangement(width=SCREEN_WIDTH_TILES, height=SCREEN_HEIGHT_TILES)
+                             for i in range(self.frames)]
+
+    def from_block(self, block, offset):
+        with EbCompressibleBlock() as compressed_block:
+            compressed_block.from_compressed_block(block=block, offset=offset)
+            self.graphics.from_block(block=compressed_block, offset=0, bpp=TILE_BPP)
+            next_offset = self.graphics_data_size
+            self.palette.from_block(block=compressed_block, offset=next_offset)
+            next_offset += self.palette.block_size()
+
+            for arrangement in self.arrangements:
+                arrangement.from_block(block=compressed_block, offset=next_offset)
+                next_offset += arrangement.block_size()
+
+    def to_block(self, block):
+        self.graphics_data_size = self.graphics.block_size(bpp=TILE_BPP, trimmed=True)
+        total_block_size = (self.graphics_data_size + self.palette.block_size() +
+                            sum(arrangement.block_size() for arrangement in self.arrangements))
+
+        with EbCompressibleBlock(total_block_size) as compressed_block:
+            self.graphics.to_block(block=compressed_block, offset=0, bpp=TILE_BPP)
+            next_offset = self.graphics_data_size
+            self.palette.to_block(block=compressed_block, offset=next_offset)
+            next_offset += self.palette.block_size()
+
+            for arrangement in self.arrangements:
+                arrangement.to_block(block=compressed_block, offset=next_offset)
+                next_offset += arrangement.block_size()
+
+            compressed_block.compress()
+            return block.allocate(data=compressed_block)
+
+    def images(self, arrangements=None):
+        return [arrangement.image(self.graphics, self.palette) for arrangement in self.arrangements]
+
+    def add_frame_from_image(self, image, frame_id):
+        self.arrangements[frame_id].from_image(image, self.graphics, self.palette, is_animation=True)
+
+
+class AnimationModule(EbModule):
+    """ Extracts non-battle animations from Earthbound. """
+    NAME = 'Animations'
+
+    # Animation Data and Animation Sequence Pointers Table
+    FREE_RANGES = [(0x0C0000, 0x0C2E18)] # TODO: This range should be changed dynamically based on the contents of the pointers table

```

Comment:
is this TODO still valid?

Reply:
Only if we want to support compiling over an already modified ROM and (I think) only when that ROM has changes made by a different editor / tool. I don't think any other modules care about that though.

Reply:
yeah, generally modules don't care about that. if this TODO is no longer valid, can you please remove it?

Reply:
Now removed.